### ISSUE TITLE
text field loosing state on `render`

### ISSUE THREAD

**Issue by perguth** [choojs/choo#579]
### Expected behavior
Input fields should retain their state regardless of `emitter.emit('render')`

### Actual behavior
Text input fields loose their state/value on rendering.

### Steps to reproduce behavior
Minimally adjusted code from the README shows the described behaviour: http://requirebin.com/?gist=829488013f4333ed528825df65c6a406

**Comment by perguth** [comment]
Yes, thanks to the friendly folks in `#choo` on freenode. I wrapped the input element in a [Nanocomponent](https://github.com/choojs/nanocomponent#es6-class-syntax) like so:
```js
let Input = class Component extends Nanocomponent {
  constructor () {
    super()
    this.state = {}
  }
  createElement (state) {
    this.state = state
    return html`
      <input onkeypress=${state.onkeypress} onfocus=${state.onfocus} onblur=${state.onblur}
      class=form-control type=text placeholder='name or mac address' data-toggle=dropdown>
    `
  }
  update () {}
}
let input = new Input()
```
<sup>([source](https://github.com/perguth/ffs-monitor/blob/58c7fd243840cd62a2c53d1e6447d923f118c7fb/browser.js#L89))</sup> and included it like so:

```js
${input.render({onkeypress: search, onfocus: showSuggestions, onblur: hideSuggestions})}
```
<sup>([source](https://github.com/perguth/ffs-monitor/blob/58c7fd243840cd62a2c53d1e6447d923f118c7fb/browser.js#L222))</sup>.

**Comment by heyitsmeuralex** [comment]
Is this a bug in nanomorph? I guess it just needs to not consider `el.value` a modified property.

**Comment by jfr3000** [comment]
is there any update on this?

my workaround in the meantime is to attach an `isSameNode` handler to the input field as documented here: https://github.com/choojs/choo#caching-dom-elements

```javascript
const inputField = input({type: "text"})
inputField.isSameNode = (target) => {
  return (target && target.nodeName && target.nodeName === 'INPUT')
}
```
